packages/mobx/src/api/annotation.ts

```diff
@@ -1,8 +1,17 @@
 import { ObservableObjectAdministration, isFunction } from "../internal"
 
+export const MAKE_CANCEL = 0
+export const MAKE_BREAK = 1
+export const MAKE_CONTINUE = 2
+
 export type Annotation = {
     annotationType_: string
-    make_(adm: ObservableObjectAdministration, key: PropertyKey): void
+    make_(
+        adm: ObservableObjectAdministration,
+        key: PropertyKey,
+        descriptor: PropertyDescriptor,
+        source: object
+    ): 0 | 1 | 2
```

Comment:
Nit: you could introduce a type for this: `export type MakeResult = 0 | 1 | 2`

Reply:
I would like to:
`export type MakeResult = MAKE_CANCEL | MAKE_BREAK | MAKE_CONTINUE`
but obviously not possible as these are runtime...

Perhaps I can do it like:
```javascript
type MakeResultCancel = 0
type MakeResultBreak = 1
type MakeResultContinue = 2

export const MAKE_CANCEL: MakeResultCancel = 0
export const MAKE_BREAK: MakeResultBreak = 1
export const MAKE_CONTINUE: MakeResultContinue = 3

export type MakeResult = MakeResultCancel | MakeResultBreak | MakeResultContinue
```
Or am I overengineering it?

Reply:
@urugator  pretty sure this will work:

```
export const MAKE_CANCEL = 0
export const MAKE_BREAK = 1
export const MAKE_CONTINUE = 2

export type MakeResult = typeof MakeResultCancel | typeof MakeResultBreak | typeof MakeResultContinue
```

But yeah we're kinda reinventing enums :) https://www.typescriptlang.org/tsconfig#references-to-const-enum-members

Reply:
Oh cool. I will use `const enum` then. I just wanted to avoid the generated object.